src/core/Rule.ml

```diff
@@ -542,6 +543,7 @@ type 'mode rule_info = {
    * Xpattern.Filename feature that integrates well with the xpatterns.
    *)
   paths : paths option;
+  product : Out.product;
```

Comment:
Can you add a comment here explaining how this field is filled?
Rule.ml is supposed to translate what we find in a rule; there is no 'product:' field in our syntax
so where does this field come from?